presto-orc/src/main/java/com/facebook/presto/orc/reader/MapFlatSelectiveStreamReader.java

```diff
@@ -236,8 +236,8 @@ public int read(int offset, int[] positions, int positionCount)
 
         outputPositions = initializeOutputPositions(outputPositions, positions, positionCount);
 
-        if (presentStream != null && keyCount == 0) {
```

Comment:
Is there a test case for read all nulls ? The previous code looks right to me. In DWRF, when all values are present, the present stream is skipped. When nulls are present, present stream must be present. Curious to know why the old block was removed.

Reply:
> Is there a test case for read all nulls ?

Yes, `TestFlatMapWriter.testMixedNullMapAndEmptyMap` covers this case. I found this case by a checksum mismatch triggered by Vader.

> When nulls are present, present stream must be present.

Right, but the old implementation in `readAllNulls` didn't care about non nulls (empty maps). 

In case of the flat map we cannot say that all values are nulls until we scan positions in the `present` stream. We could do it for the regular map because it has additional `length` stream, and `length == null && present !=null` meant that all values are null.